cpp/src/parquet/column_writer.h

```diff
@@ -87,8 +88,9 @@ class PARQUET_EXPORT PageWriter {
 
   static std::unique_ptr<PageWriter> Open(
       std::shared_ptr<ArrowOutputStream> sink, Compression::type codec,
-      int compression_level, ColumnChunkMetaDataBuilder* metadata,
-      int16_t row_group_ordinal = -1, int16_t column_chunk_ordinal = -1,
+      const std::shared_ptr<CodecOptions>& codec_options,
```

Comment:
This should be put to the end for capability

Reply:
Also, unless the `PageWriter` needs to keep ownership of the codec options, this should simply be `const CodecOptions& = {}`.

Reply:
In addition to the comments above, should be consolidate `Compression::type codec` and `const std::shared_ptr<CodecOptions>& codec_options`? codec_options should know codec.